design/ozone.md

```diff
@@ -0,0 +1,258 @@
+
+The version of Ozone can be decoupled from the version of the Operator similar to the [CSI version decoupling](https://github.com/rook/rook/blob/master/design/decouple-ceph-version.md). The default version will be defined by the rook operator version but other version can be used based on CRD entry. 
+
+### Kubernetes resource generation
+
+Ozone provides examples K8s resources files and transformation definitions using [flekszible](https://github.com/elek/flekszible). This is a composition based k8s resource generator where the transformations can be reused.
+
+Let's say we have a simple kubernetes resource:
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: datanode
+  labels:
+    app.kubernetes.io/component: ozone
+spec:
+  selector:
+    matchLabels:
+      app: ozone
+      component: datanode
+  serviceName: datanode
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: ozone
+        component: datanode
+    spec:
+      containers:
+      - name: datanode
+        image: "apache/hadoop:0.5.0-SNAPSHOT"
+        args: ["ozone","datanode"]
+        ports:
+           - containerPort: 9870
+             name: rpc
+```
+
+Prometheus annotations can be added by a simple rule:
+
+```yaml
+type: Add
+path: ["spec","template","spec","metadata","annotations"]
+trigger:
+   metadata:
+      name: datanode
+value:
+   prometheus.io/scrape: "true"
+   prometheus.io/port: "9882"
+   prometheus.io/path: "/prom"
+```
+
+This transformation can be applied to any of the k8s resources.
+
+Until now it is very similar to Kustomize, but here we have more predefined transformations (NodePort service generator, value replacer).
+
+Even better the transformations can be reused. The previous definition can be reused by putting it to the right directory:
+
+```yaml
+name: ozone/prometheus
+description: Enable prometheus monitoring in Ozone
+---
+- type: Add
+  path: ["spec","template","spec","metadata","annotations"]
+  trigger:
+     metadata:
+        name: datanode
+  value:
+     prometheus.io/scrape: "true"
+     prometheus.io/port: "9882"
+     prometheus.io/path: "/prom"
+```
+
+After this step, you can use it as
+
+```yaml
+type: ozone/prometheus
+```
+
+Or from golang API: 
+
+```golang
+
+context, err := Initialize("../testdata/api")
+if err != nil {
+   return err
+}
+
+err = context.AppendCustomProcessor("ozone/prometheus", make(map[string]string))
+if err != nil {
+   return err
+}
+
+context.Render()
+
+s, err := context.Resources()[0].Content.ToString()
+if err != nil {
+   return err
+}
+
+assert.Contains(t, s, "prometheus.io/scrape: \"true\"")
+```
+
+### Create k8s resource from operator
+
+One easy way to integrate the existing resource generation of Ozone is to use it from golang API. This is similar to the approach of [ceph csi](https://github.com/rook/rook/tree/master/pkg/operator/ceph/csi) which reads the k8s template files from the disk.
```

Comment:
`ceph csi` -> `Ceph CSI`

Reply:
Thanks, fixed (with a few other occurrence of the ceph->Ceph)